dbaas_nfsaas/faas_provider.py

```diff
@@ -9,27 +9,50 @@
 
 class Provider(object):
 
-    def __init__(self, dbaas_api, host_class):
+    def __init__(self, dbaas_api, host_class, group_klass):
         self.dbaas_api = dbaas_api
         self.client = Client(
             authurl=dbaas_api.endpoint, user=dbaas_api.user,
             key=dbaas_api.password, tenant_name=dbaas_api.tenant_name,
             insecure=dbaas_api.is_secure
         )
         self.host_class = host_class
+        self.group_klass = group_klass
+
+    def _get_or_create_infra_group(self, host):
```

Comment:
In this method get or create, but dont create anything or u fogot the group.save() ? if u save that maybe get_or_update will be better :D